configurator/src/main/java/com/devonfw/tools/ide/migrator/xml/XmlCommentNodeReplacement.java

```diff
@@ -0,0 +1,51 @@
+package com.devonfw.tools.ide.migrator.xml;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/**
+ * Implementation of {@link AbstractXmlMigration} for a simple "comment node" replacement
+ */
+public class XmlCommentNodeReplacement extends AbstractXmlMigration {
+
+  private String search;
+
+  private String replacement;
+
+  /**
+   * The constructor.
+   *
+   * @param search the {@link String} to search for.
+   * @param replacement the replacement for the given {@code search} {@link String}.
+   */
+  public XmlCommentNodeReplacement(String search, String replacement) {
+
+    super();
+    this.search = search;
+    this.replacement = replacement;
+  }
+
+  @Override
+  public boolean migrateXml(Document xml) throws Exception {
+
+    return migrateXmlCommentNode(xml.getChildNodes());
+  }
+
+  private boolean migrateXmlCommentNode(NodeList childNodes) {
+
+    boolean updated = false;
+
+    for (int i = 0; i < childNodes.getLength(); i++) {
+      Node node = childNodes.item(i);
+      short nodeType = node.getNodeType();
+      if (nodeType == Node.COMMENT_NODE) {
+        if (this.search.equals(node.getNodeValue().trim())) {
```

Comment:
a regex or an substring match/replace would IMHO make more sense than an exact literal match (that makes sense for tag-names where you only want to replace `<tag>` and not accidentally also `<tagname>`.